### PR TITLE
Update to CLDR 44, adding top-to-bottom character direction

### LINKED ISSUE #70: Add override for skr locale (text direction RTL)
Saraiki (skr) is currently not available in CLDR, but it's completely localized in Firefox.

Adding it to CLDR is going to take a long time, so ti would be ideal to have a work-around here (set the direction for the locale, overriding CLDR data).

### PR DESCRIPTION
**Pull request by eemeli**
Fixes #70 

This updates the CLDR data to the latest release 44.0.1. We're primarily interested in this for Firefox as that release includes data for Saraiki, which we're currently rendering with LTR rather than RTL directionality.

The data updates since v37 require two adjustments to the locale direction logic:
1. Some of the added locales (bal-Latn, ks-Deva, sd-Deva) are LTR even though their parent languages default to an RTL script. So now there's a `SCRIPTS_CHARACTER_DIRECTION_LTR` table that's checked if the locale id includes a script. Also, the assumption about this never happening was removed.
2. One added locale (mn-Mong) uses a "top-to-bottom" character direction. To support that, the `CharacterDirection::TTB` value needs to be added to the enum.

The data was updated as follows:
```sh
cd unic-langid-impl/data

# Start from a clean slate
rm -r cldr-misc-full

# Fetch the new data via npm
npm install --no-save cldr-core@latest cldr-misc-full@latest
mv node_modules/cldr-core/supplemental/likelySubtags.json .
mv node_modules/cldr-misc-full .

# Drop unnecessary files
rm -r node_modules
find cldr-misc-full/main -type f ! -name layout.json -delete
rm cldr-misc-full/bower.json
```

### CODE REVIEW

**Comment by eemeli** [comment]
@zbraniecki Let me know if you'd like to have the data updates split into its own commit. As is, the "3140 files changed" make the GitHub interface somewhat cumbersome to review.

**Comment by zbraniecki** [comment]
yeah, please, separate to two:
* Change support for script direction / Add TTB /  Update any scripts
* Update to new CLDR

**Comment by flodolo** [comment]
Passing by comment: zibi's email is updated in the README, but needs to be updated in each crate's TOML (still mozcom).
I guess that can be taken care as part of the version bump.

**Comment by eemeli** [comment]
@zbraniecki Refactored to three because that seemed simpler:
1. Add support for new features, without changing observable behavior with CLDR 37
2. Add CLDR 44 data
3. Update tables & tests using new data

File: unic-langid-impl/src/bin/generate_layout.rs [hunk 1]
```diff
@@ -28,33 +28,27 @@ fn langid_to_direction_map(path: &str) -> HashMap<LanguageIdentifier, CharacterD
...
         result.insert(langid, character_order);
     }
     result
 }
 
-fn check_all_variants_rtl(
+fn check_all_script_variants(
     map: &HashMap<LanguageIdentifier, CharacterDirection>,
-    lang: Option<Language>,
+    exp_dir: &CharacterDirection,
```
Comment (zbraniecki) [inline unic-langid-impl/src/bin/generate_layout.rs]: `nit`: please, make `CharacterDirection` `Copy` don't use a reference here.

File: unic-langid-impl/src/bin/generate_layout.rs [hunk 2]
```diff
@@ -63,68 +57,98 @@ fn main() {
...
+            exp_dir = &CharacterDirection::LTR;
+        } else if dir == &CharacterDirection::RTL {
+            scripts = &mut scripts_rtl;
+            exp_dir = &CharacterDirection::RTL;
+
+            let lang = langid.language;
+            langs_rtl.insert(lang);
+        } else {
+            scripts = &mut scripts_ttb;
+            exp_dir = &CharacterDirection::TTB;
```
Comment (zbraniecki) [inline unic-langid-impl/src/bin/generate_layout.rs]: `nit`: Rust supports `if/else` as an expression. You can do:

```rust
let (scripts, exp_dir) = if ... {
  (&mut scripts_ltr, CharacterDirection::LTR)
} else {
  (&mut scripts_rtl, CharacterDirection::RTL)
};
```

or even better:

```rust
let (scripts, exp_dir) = match dir {
  CharacterDirection::LTR => (&mut scripts_ltr, CharacterDirecton::LTR),
  CharacterDirection::RTL => {
      langs_rtl.insert(langid.language);
      (&mut scripts_rtl, CharacterDirection::RTL)
  },
  ...
}
```

**Review by zbraniecki — APPROVED** [review]
lgtm.

**Review by zbraniecki — APPROVED** [review]

File: unic-langid-impl/src/bin/generate_layout.rs (hunk 2 above)
Comment (eemeli) [reply unic-langid-impl/src/bin/generate_layout.rs]: Done. Actually, the `exp_dir` turned out to be unnecessary and got factored out.

File: unic-langid-impl/src/bin/generate_layout.rs (hunk 1 above)
Comment (eemeli) [reply unic-langid-impl/src/bin/generate_layout.rs]: I tried, but `CharacterDirection` does not implement the `Copy` trait.

Comment (zbraniecki) [reply unic-langid-impl/src/bin/generate_layout.rs]: Add it:
```rust
#[derive(Copy, ...)]
enum CharacterDirection { ... }
```

**Comment by zbraniecki** [comment]
Apologies for slow response. Released!

Let me know if you need to bump to syn 2.0, I merged it after the release. 

Also, happy to add someone from Mozilla as a contributor so that you can release yourself!

**Pull request merged**